playground/pspin/pspin.py

```diff
@@ -0,0 +1,62 @@
+import matplotlib
+import numpy as np
+import matplotlib.pyplot as plt
+import scipy
+from scipy.misc import comb, factorial
+from pele.optimize._quench import lbfgs_cpp, modifiedfire_cpp
+from pele.potentials import MeanFieldPSpinSpherical
+from joblib import Parallel, delayed
+
+
+p=3
+nspins=20
+interactions = np.ones(np.power(nspins,p))
+coords = np.ones(nspins)
+pot = MeanFieldPSpinSpherical(interactions, nspins, p, tol=1e-6)
+e = pot.getEnergy(coords)
+assert e == -comb(nspins,p)/nspins
+
+interactions = np.random.normal(0, np.sqrt(factorial(p)), [nspins for i in xrange(p)])
+#interactions =  (interactions + interactions.transpose())/2
+for i in xrange(nspins):
+    for j in xrange(i, nspins):
+        for k in xrange(j, nspins):
+            interactions[k][i][j] = interactions[i][j][k]
+            interactions[k][j][i] = interactions[i][j][k]
+            interactions[j][k][i] = interactions[i][j][k]
+            interactions[i][k][j] = interactions[i][j][k]
+            interactions[j][i][k] = interactions[i][j][k]
```

Comment:
The sum of two normally distributed random variable is again random, just with different mean and variance.  https://en.wikipedia.org/wiki/Sum_of_normally_distributed_random_variables.   Therefore you might be able to do

    interactions = interactions + interactions.transpose()

to avoid the massively slow python tripple loop.   Although it will probabaly be more complictated for p>2.  Maybe something like